### DISCUSSION TITLE
Pause at Layer is Unloading Filament

### DISCUSSION THREAD

**Discussion by BlueWings172** [Jyers/Marlin#1840] [Q&A] [answered]
Hello

I've been struggling with the pause at layer script in Cura. Printer pauses at indicated layer number and when the printer resumes, I noticed the filament does not come out of the nozzle. Turned out that once I resume the print, the printer for some reason retracts few centimeters of filament and continues to print while the filament end is in the heat break. I have this 15-hour print that is I need to pause twice to put some nuts in and this has been driving me crazy for 2 days.

Would appreciate any help.

Note: i have a MS Direct Drive setup.

**Comment by BlueWings172** [comment, marked as answer]
this has been solved.

**Reply by Cal-LF** [reply]
@BlueWings172 Can you tell the solution for others to refer if similar issue?

**Reply by Cal-LF** [reply]
@tititopher68-dev I found that 0 length is manual load/unload. My lengths are 455 and 450. 
Default is 0 for each.

**Reply by BlueWings172** [reply]
I wish I have an answer than can benefit others. the things is, I'm not trying to unload filament, just stop at a certain layer number to insert nuts then resume the print. I can't understand why the printer would retract few dozen mm of filament just before resuming the print. 

What I did is the nuclear option: Creality firmware instead of Jyers + old Octoprint copy I have on an SD card + Creality slicer instead of Cura. So basically changed all software involved. Been banging my head against the wall for 2 days and I had to resort to that.
